### ISSUE TITLE
URL Detection and Selection

### ISSUE THREAD

**Issue by codythegreat** [codythegreat/dss#38]
### URLs

URL handling will be done with markdown style formatting `[title](www.webpage.com)`

When dss detects the above format, it'll store the website url to the slide struct, and format the slide to print `[n]title` whereas `n` is the link number (in the case of a single link, this would be `0`. 

If a user presses `l` followed by the link number the webpage will be opened in the default browser.

**Comment by amagura** [comment]
We could use something like [cmark](https://github.com/commonmark/cmark) to parse each slide.

**Comment by codythegreat** [comment]
> We could use something like cmark to parse each slide.

Interesting. I'll read up on that repo some more. Perhaps that can aid us in the markdown parsing mode as well.

**Comment by codythegreat** [comment]
commit e342d937d5d04b24ff5faeba592b0f769418561e and ee321f909bbd285b59010e9a289e438d1e3d5194 added some basic logic for handling parsing, printing, and selecting links. This code is still very rudimentary, but currently if you have a link in the following format: `[Google](https://www.google.com)` you can actually select and open the link in the default browser.

Things that I'd like to change:

+ links are handled as an array (of len 10) of char arrays in the `slide` struct. This is not optimal as we'd like for any number of links to be accommodated. I'd like to create a new type like this:
``` 
// first instance of link stored in slide struct,
struct link {
    int index;
    char *title;
    char *url;
    link *next;
}
```
+ Currently the format for links is pretty strict. If the user forgets to include `http[s]://` in their link the browser simply won't open. If the user forgets to include this, we'd like for the logic to automatically add it. this should be an easy fix.
+ I'd like for the parser to remove the markdown style raw text `[title](link)` from the line and replace it with `title[index]` / `Google[0]`

**Comment by codythegreat** [comment]
So I just pushed 7 commits all related to URL parsing. Here is a quick synopsis of changes:

+ I didn't like using an array of links, so I created the `mdlink` struct that is a linked list. the slide will either store the first link, or NULL if a link doesn't exist. each link has an index (starting at `0`) that is callable by pressing `l[index]` when on the slide containing the link
+ title is now being parsed properly. It is currently not being used however.

One more thing for the todo list:

+ free the links upon opening new file/exiting dss

**Comment by codythegreat** [comment]
Outside of these two features:
> + I'd like for the parser to remove the markdown style raw text [title](link) from the line and replace it with title[index] / Google[0]
> + I'd also like to move to something more robust for the parsing portion of the logic (possibly cmark?) that can hopefully be extended to more markdown style parsing.

I'd consider link parsing to be mostly complete. so I'll go ahead and close this issue. These features will likely be reopened as separate issues. I'm hoping that the first issues above can be implemented before 0.3.0 release. We will see.
